File: rust/tak.rs
use std::env;

fn tak(x: i32, y: i32, z: i32) -> i32 {
    if y < x {
        return tak(tak(x - 1, y, z), tak(y - 1, z, x), tak(z - 1, x, y));
    }
    z
}

fn main() {
    let args: Vec<String> = env::args().collect();

    let x: i32 = args[1].parse().unwrap();
    let y: i32 = args[2].parse().unwrap();
    let z: i32 = args[3].parse().unwrap();

    println!("{}", tak(x, y, z));
}

File: csharp/tak.cs
class TakProgram
{
	static int Tak(int x, int y, int z)
	{
		if (y < x)
		{
			return Tak(
				Tak(x - 1, y, z),
				Tak(y - 1, z, x),
				Tak(z - 1, x, y)
			);
		}
		else
		{
			return z;
		}
	}

	static void Main(string[] args)
	{
		int x = int.Parse(args[0]);
		int y = int.Parse(args[1]);
		int z = int.Parse(args[2]);
		Console.WriteLine(Tak(x, y, z));
	}
}
